### PR TITLE
Fix #155 Add OpenIDConnect

### LINKED ISSUE #155: Add support for OpenID Connect (OIDC)?
Are there plans to support [OpenID Connect](http://openid.net/connect/) (aka OIDC)?

OpenID Connect is an authentication layer built on top of the OAuth2 authorization framework.  Now that OIDC has replaced OpenID and is based upon the popular OAuth2 framework, OIDC adoption is growing quickly and poised to become the industry standard for authentication and authorization.

I see that yii2-authclient supports both OpenID and OAuth2 so it should be fairly easy to support OpenID Connect. Unfortunately, I don't have the skills to implement this. There is a third-party extension "[yii2-openidconnect](https://github.com/CommonMode/yii2-openidconnect)", but I'm not sure it is production ready.  Its code may be a good starting point or perhaps it could be merged.

### Some background on OpenID Connect:
- [http://openid.net/connect/](http://openid.net/connect/)
- [http://openid.net/connect/faq/](http://openid.net/connect/faq/)
- [http://en.wikipedia.org/wiki/OpenID_Connect](http://en.wikipedia.org/wiki/OpenID_Connect)

### Some major providers supporting OpenID Connect:
- [Google](https://developers.google.com/identity/protocols/OpenIDConnect)
- [Microsoft Azure](https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-protocols-openid-connect-code)
- [Amazon Web Services](http://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html)
- [Ping Identity](https://www.pingidentity.com/en/resources/articles/openid-connect.html)
- [Gigya Identity Management](https://developers.gigya.com/display/GD/OpenID+Connect)
- [Janrain Identity & Access Management](http://www.janrain.com/blog/janrain-supports-openid-connect/) 
- [Gluu Identity & Access Management](https://www.gluu.org/resources/documents/standards/openid-connect/)
- [WSO2 Identity & Access Management](https://docs.wso2.com/display/IS510/OAuth2-OpenID+Connect)
- [ForgeRock Access Management](https://www.forgerock.com/blog/openam-now-openid-certified/)
- [Centrify Identity Server](https://docs.centrify.com/en/centrify/appref/index.html#page/cloudhelp/gen/saas_appref_openid.html)
- [IBM Websphere](https://www.ibm.com/support/knowledgecenter/SSD28V_8.5.5/com.ibm.websphere.wlp.core.doc/ae/cwlp_openid_connect.html)
- [Paypal](https://developer.paypal.com/docs/integration/direct/identity/log-in-with-paypal/)
- [Salesforce](https://developer.salesforce.com/page/Inside_OpenID_Connect_on_Force.com)
- [VMware Authentication Services](https://vmware.github.io/lightwave/)

Note that this started as [a topic on the Yii forums](http://www.yiiframework.com/forum/index.php/topic/74594-support-for-openid-connect-oidc/)

### PR DESCRIPTION
**Pull request by klimov-paul**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #155

### CODE REVIEW

**Comment by samdark** [comment]
Code looks good. Haven't actually tested it yet.

File: OpenIdConnect.php [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+            throw new InvalidConfigException('Unable to authenticate request: none of following auth methods is suported: ' . implode(', ', $supportedAuthMethods));
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function defaultReturnUrl()
+    {
+        $params = $_GET;
```
Comment (cebe) [inline OpenIdConnect.php]: better use `Yii::$app->request->get()` ?

**Comment by cebe** [comment]
looks good overall.

File: OpenIdConnect.php (hunk 1 above)
Comment (samdark) [inline OpenIdConnect.php]: Yes. Absolutely.

**Pull request merged**